### DISCUSSION TITLE
The first character is not visible in the parameter input screen

### DISCUSSION THREAD

**Discussion by imishinist** [knqyf263/pet#305] [General]
When I run `pet search` or `pet clip`, the first character is not visible in the parameter entry screen.
When I press `enter`, the invisible characters are also entered correctly, so perhaps they are just not visible.

Here is the tape file and gif when reproduced by [vhs](https://github.com/charmbracelet/vhs).

![out](https://github.com/knqyf263/pet/assets/5667851/8f264c34-1fb5-428b-83ab-e5394f452f58)

<details><summary>pet.tape</summary><p>

```
Type "pet search"
Sleep 3s
Enter

Sleep 3s
Enter

Type "1.1.1.1"
Sleep 3s
Enter

Sleep 10s
```
</p></details>

To see where the problem is occurring, I tried building with the following patch.

<details><summary>patch</summary><p>

```diff
diff --git a/dialog/view.go b/dialog/view.go
index ff75f11..ce558ae 100644
--- a/dialog/view.go
+++ b/dialog/view.go
@@ -120,6 +120,13 @@ func GenerateParamsLayout(params [][2]string, command string) {
 	maxX, maxY := g.Size()
 	leftX := (maxX / 2) - (maxX / 3)
 	rightX := (maxX / 2) + (maxX / 3)
+	g.SelBgColor = gocui.ColorCyan
+	g.BgColor = gocui.ColorGreen
+	g.SelFgColor = gocui.ColorBlue
+	g.FgColor = gocui.ColorYellow
+	g.FrameColor = gocui.ColorRed
+	g.SelFrameColor = gocui.ColorMagenta
+
 
 	generateSingleParameterView(g, "Command(TAB => Select next, ENTER => Execute command):",
 		command, []int{leftX, maxY / 10, rightX, maxY/10 + 5}, false)
```
</p></details>

![out](https://github.com/knqyf263/pet/assets/5667851/98402931-b2f9-4a64-90e0-0d3e25c39c6c)

Could this be a bug?

Environments
- macOS Sonoma 14.4.1
- version: 0.8.4
- go version: go version go1.21.5 darwin/arm64
- HEAD: a93ab023e8c82a67bc7a20d55bfa5df177d7b9b5

**Comment by imishinist** [comment]
Apparently, the problem is due to the use of double-byte characters in the frames.

The problem stopped occurring when I wrote the following statement.

```go
g.ASCII = true
```

**Comment by RamiAwar** [comment]
Very interesting, it doesn't happen for me locally on the same OS and versions 😅 

I'm not sure if changing this to ASCII for everyone would have any impact or not. Any ideas?

**Reply by imishinist** [reply]
I suspect this problem is due to a bug in gocui.

I have created an issue that includes the code to reproduce it.
https://github.com/awesome-gocui/gocui/issues/131

If it is a bug in gocui, wouldn't it be a good idea to fix it and then import a newer version of gocui?

**Reply by RamiAwar** [reply]
Ahh I see, yeah it does look like a gocui bug then! Thanks for putting in all the effort and coding up a reproduction! I'm sure that took some time 😄 

Yeah we can definitely update gocui once this is fixed! 🙂
